test/data/manifests/fedora-coreos-container.mpp.yaml

```diff
@@ -201,42 +201,29 @@ pipelines:
             - from: input://tree/
               to: mount://root/
         devices:
-          efi:
+          disk:
             type: org.osbuild.loopback
             options:
               filename: disk.img
-              start:
-                mpp-format-int: '{image.layout[''EFI-SYSTEM''].start}'
-              size:
-                mpp-format-int: '{image.layout[''EFI-SYSTEM''].size}'
-          boot:
-            type: org.osbuild.loopback
-            options:
-              filename: disk.img
-              start:
-                mpp-format-int: '{image.layout[''boot''].start}'
-              size:
-                mpp-format-int: '{image.layout[''boot''].size}'
-          root:
-            type: org.osbuild.loopback
-            options:
-              filename: disk.img
-              start:
-                mpp-format-int: '{image.layout[''root''].start}'
-              size:
-                mpp-format-int: '{image.layout[''root''].size}'
+              partscan: true
         mounts:
           - name: root
             type: org.osbuild.xfs
-            source: root
+            source: disk
+            partition:
+              mpp-format-int: '{image.layout[''root''].partnum}'
             target: /
           - name: boot
             type: org.osbuild.ext4
-            source: boot
+            source: disk
+            partition:
+              mpp-format-int: '{image.layout[''boot''].partnum}'
             target: /boot
           - name: efi
             type: org.osbuild.fat
-            source: efi
+            source: disk
+            partition:
+              mpp-format-int: '{image.layout[''EFI-SYSTEM''].partnum}'
```

Comment:
Fwiw, I wonder if we can do the same simplification  the bits that do org.osbuild.mkfs.*, i.e. replace "start/size" there with `partition: mpp-format-int: '{image.layout[''xxx''].partnum}'` ?

Reply:
probably!